File: lcode/tracholar/sm.cc
#include<stdio.h>
#include<stdlib.h>

int main(){
  int n;
  scanf("%d", &n);
  if(n < 3){
    printf("0\n");
  }
  int* p = new int[n];

  for(int i=0; i<n; i++){
    scanf("%d", &p[i]);
  }

  int sum = 0;
  int left = 0;
  int right = 0;
  int A;
  for(int i = 1; i < n-1; i++){
    A = p[i];
    left = 0;
    right = 0;
    for(int j=0; j<i; j++){
      if(p[j] < A){
        left++;
      }
    }
    for(int j=i+1; j<n; j++){
      if(p[j] < A){
        right++;
      }
    }

    //printf("A=%d,left=%d,right=%d\n", A, left, right);
    sum += left * right;
  }

  printf("%d\n", sum);
}

File: lcode/tracholar/sm.py

def main():
    n = int(input().strip())
    if n < 3:
        print(0)
        return

    line = input().strip()
    arr = [int(x.strip()) for x in line.split(' ') if len(x.strip())>0]

    P = 0
    for j, A in enumerate(arr):
        if j in (0, n-1):
            continue
        left = sum(1 for x in arr[:j] if x < A)
        right = sum(1 for x in arr[j+1:] if x < A)

        P += left * right

    print(P)

main()

